scripts/Client.js

```diff
@@ -39,66 +40,67 @@ function getRow(row) {
     props.ContactOption = values[0][5];
     props.CalleeNote = values[0][6];
     props.ContactDate = "";
+    //console.log((values[0]));
 //*/
     
-  }).getRow(SPREADSHEET_ID, row);
+  }).getRow(SPREADSHEET_ID, row, campaign);
 }
 
 
 
-
-function getRange(row) {
+function getRange(row) { 
   
   var props = app.currentPage.properties;
   var name = props.Name.toUpperCase();
-
+   
   google.script.run.withSuccessHandler(function(result1) {
     google.script.run.withSuccessHandler(function(result2) {
 //      console.log(result1);      
 //      console.log(result2);
       
       CURR_ROW = result1;
-      END_ROW = result2;
+      START_ROW = result2.start;
+      END_ROW = result2.end;
       row = parseInt(result1);
       var e = parseInt(result2);
-//      console.log("curr: " + CURR_ROW + " END: " + END_ROW + " row: " + row);
-      getRow(CURR_ROW);
+      console.log("start: " + START_ROW + " END: " + END_ROW + " row: " + row + "current: "+CURR_ROW);
+      getRow(CURR_ROW, props.ChosenCampaign); 
```

Comment:
having the global campaign would be faster than doing `props.ChosenCampaign` because it'd be already initialized while with this line, `app` would call `currentPage` which would call `properties` which would then call `ChosenCampaing`. same thing for lines 69 and 70